Sqlite provider added + fixes in sql_server_user provider

rdoorn opened the pull request:
* Add SQLite provider (drop and query support)
* Allow to pass options to the connection string for all databases for setting initial settings
* fix in sql_server_user provider where a unused variable was used (sql_role -> sql_sys_role attribute mismatch)

tas50 commented on README.md:
```diff
@@ -249,6 +256,22 @@ postgresql_database 'vacuum databases' do
+# Create, Insert, Query a SQLite database
+# Note that inserting anything in to the database will create it automaticly.
+sqlite_database 'mr_softie' do
+  connection '/path/to/database.db3'
```
Can we make this database so it matches the postgresql provider and better describes what the attribute is?

rdoorn replied:
```database_name``` seems to be the common database name object, postgresql allows the ```database``` part in the connection hash to override database_name. as such i think that database_name is the better object to use. I'll make the adjustment to database_name, unless you have strong objections.

tas50 replied:
Name probably does make more sense.  As we try to abstract the various DBs we should aim to make it as common as possible.  They're all very different technologies, so that's not 100% possible, but where it can be done we should.

tas50 commented on libraries/provider_database_mysql_user.rb:
```diff
@@ -105,7 +105,7 @@ def ishash?
             test_sql_results = test_client.query test_sql
 
-            incorrect_privs = true if test_sql_results.empty?
+            incorrect_privs = true if test_sql_results.find { |k| k['User'] }.nil?
```
Was this meant to be included?

rdoorn replied:
this is unrelated to the original pull request, but it fixes the bug in the mysql provider which causes the travis build to fail, i figured i might as well fix this while i'm at it. so the travis build is successful now.

tas50 commented on libraries/provider_database_sql_server_user.rb:
```diff
@@ -106,12 +106,22 @@ def action_alter_sys_roles
           Chef::Log.info("SQL Server Version: #{server_version.inspect}")
           db.execute('USE [master]').do
           @new_resource.sql_sys_roles.each do |sql_sys_role, role_action|
-            alter_statement = if server_version < '11.00.0000.00'
```
I think you reverted the SQL Server changes that went in recently with this commit

rdoorn replied:
indeed i did, I will correct this and put back the more recent code.

tas50 commented on recipes/sqlite.rb:
```diff
@@ -0,0 +1,21 @@
+
+chef_gem 'sqlite3' do
+  compile_time true
+end
```
You might want to take a look at this new method I added to the aws cookbook to dynamically install the gem when the provider attempts to and fails to do the require for the gem.

https://github.com/chef-cookbooks/aws/pull/231

rdoorn commented:
Hi @tas50, I've corrected the requested items, and added some tests too. please let me know if there are any additional things that need taken care of. Thanks!

chef-supermarket commented:
Hi. I am an automated pull request bot named Curry. There are commits in this pull request whose authors are not yet authorized to contribute to Chef Software, Inc. projects or are using a non-GitHub verified email address. To become authorized to contribute, you will need to sign the Contributor License Agreement (CLA) as an individual or on behalf of your company. [You can read more on Chef's blog.](https://www.getchef.com/blog/2014/06/23/changes-to-the-contributor-license-agreement-process)

## GitHub Users Who Are Not Authorized To Contribute

The following GitHub users do not appear to have signed a CLA:

* @enzor

[Please sign the CLA here.](https://supermarket.chef.io/become-a-contributor)

chef-supermarket commented:
Hi. Your friendly Curry bot here. Just letting you know that all commit authors have become authorized to contribute. I have added the "Signed CLA" label to this issue so it can easily be found in the future.

rdoorn commented:
i'll recreate this pull request, as the rebase made it a mess
